### PR TITLE
Junos:   bgp_address_family resource module added

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>
resolve:
-  https://github.com/ansible-collections/junipernetworks.junos/issues/147
-   https://github.com/ansible-collections/junipernetworks.junos/issues/146
model: https://github.com/ansible-network/resource_module_models/pull/117
##### ISSUE TYPE

 
- New Module Pull Request

##### COMPONENT NAME

junos_bgp_address_family
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/junos_bgp_address_family.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [
+        ("state", "merged", ("config",)),
+        ("state", "replaced", ("config",)),
+        ("state", "rendered", ("config",)),
```
Comment (rohitthakur2590) [inline plugins/modules/junos_bgp_address_family.py:400]: need to update for overridden.

File: plugins/modules/junos_bgp_address_family.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - parsed
+    - gathered
+    - rendered
+    default: merged
+"""
+EXAMPLES = """
```
Comment (rohitthakur2590) [inline plugins/modules/junos_bgp_address_family.py:348]: need to update examples

File: plugins/modules/junos_bgp_address_family.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+  returned: when changed
+  type: dict
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
```
Comment (rohitthakur2590) [inline plugins/modules/junos_bgp_address_family.py:378]: xml mentions

File: plugins/module_utils/network/junos/argspec/bgp_address_family/bgp_address_family.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    """The arg spec for the junos_bgp_address_family module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {
+        "config": {
+            "options": {
+                # 'as_number': {'type': 'str'},
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/junos/argspec/bgp_address_family/bgp_address_family.py:42]: remove commented code

**Comment by ashwini-mhatre** [comment]
LGTM

**Pull request merged**